### ISSUE TITLE
'shrinkage' values of clf.booster_.dump_model() for 'class_weight'='balanced' case

### ISSUE THREAD

**Issue by arsenzaryan** [Microsoft/LightGBM#1802]
For a binary classification task, in the dictionary returned by clf.booster_.dump_model() there is the following issue:
In it's tree_info we have a list containing info on all trees in the model. The shrinkage parameter for the first tree is 1 and for all the rest it is equal to the learning_rate. For the binary classification task, when the class_weight is not set to balanced, it is all correct, the contribution of the first tree is much bigger (~2 orders of magnitude, when learning rate is 0.03) than the contribution of all the other trees, which means that the leaf_values of all the trees except the first one are already multiplied by shrinkage value, as also the model dump suggest.

But, for the same binary classification, when class_weight is set to balanced, the leaf values for all the trees are of the same order, but the shrinkage for the first tree is still 1. It seems that the contribution of all the trees are calculated as corrections on 0 value for the class_weight=balanced case, therefore the shrinkage value for the first tree should not be equal to 1 (as the model dump suggests) but should be equal to the learning rate as for all the remaining trees.

Example:
for one concrete test data point the individual tree predictions are:
1)for class_weight=None model: [23.236, 1.863, 0.773, 0.638, etc]
2)for class_weight=balanced model: [0.251, 0.208, 0.202, 0.199, etc]

it seems that for (1) the first tree has no shrinkage, while for (2) the first tree has shrinkage, and therefore in model dump the first tree shouldn't have shrinkage=1, but rather should have shrinkage=learning_rate like for all the remaining trees.

**Comment by guolinke** [comment]
Please set boost_from_avergaeand try again

**Comment by guolinke** [comment]
https://github.com/Microsoft/LightGBM/issues/1041
https://github.com/Microsoft/LightGBM/issues/1514#issuecomment-404689849

**Comment by arsenzaryan** [comment]
Ok, thank you for the links, the discussion #1514 looks the most relevant. If the reply there 'The first tree behavior is normal. It adds the avg to the first tree(after shrinkage), and then change it shrinkage rate to 1.' is correct, then it answers my question.
